src/index.d.ts

```diff
@@ -0,0 +1,825 @@
+// Type definitions for node-libtidy v0.3
+// Project: node-libtidy
+// Definitions by: Wang Guan <momocraft@gmail.com> / https://github.com/jokester
+
+export const tidyBuffer: TidyBufferStatic
+export const TidyDoc: TidyDocConstructor
+export const TidyOption: TidyOptionConstructor
+export const compat: TidyCompat
+/// <reference types="node" />
+
+/**
+ *  Callback convention: the result
+ */
+interface TidyResult {
+    /**
+     * errlog contains the error messages generated during the run,
+     * formatted as a string including a trailing newline.
+     */
+    errlog?: string
+    /**
+     * output contains the output buffer if output was generated.
+     * The property is unset if generating output was not part of the method
+     * in question, or null if no output was generated due to errors.
+     */
+    output?: Buffer
+}
+
+/**
+ * Callback convention: the signerature used in async APIs
+ */
+interface TidyCallback {
+    (err: Error | null, res: TidyResult | null): void
+}
+
+/**
+ * High-level functions automate the most common workflows.
+ *
+ * The document is assumed to be a buffer or a string.
+ * Anything else will be converted to a string and then
+ * turned into a buffer.
+ */
+interface TidyBufferStatic {
+    (document: string | Buffer, options: Generated.OptionDict,
+        callback: TidyCallback): void
+
+    (document: string | Buffer, callback: TidyCallback): void
+}
+
+interface TidyOption {
+    readonly name: string
+    readonly type: "boolean" | "integer" | "doctype" | "string"
+    readonly readOnly: boolean
+    readonly pickList: string[]
+    readonly id: number
+    toString(): string
+}
+
+/**
+ * TidyOption type and constructor (not for public use)
+ */
```

Comment:
If it's not for public use, then better leave this out. I had included it in the API docs for two reasons: to explain that TidyOption *has* a constructor, which can be used for instanceof checks and so on, and to explicitely state that the constructor should not be used. Not including it in the TS library would strengthen that message I think.